### ISSUE TITLE
Using @javax.transaction.Transactional fails on GraalVM

### ISSUE THREAD

**Issue by morki** [micronaut-projects/micronaut-sql#186]
I have some problem / misunderstanding with Micronaut Data Transactions and JOOQ.

### Steps to Reproduce

```kotlin
@Controller("/test")
open class IndexController(private val dslContext: DSLContext) {
    @Get("/test1")
    @Transactional
    open fun update1(): HttpResponse {
        // etc...
    }

    @Get("/test2")
    @Transactional
    open fun update2(): HttpResponse {
        // etc...
    }
}
```

### Expected Behaviour

When I remove `@Transactional` annotation from `update1()`, only `update2()` should run in transaction.

### Actual Behaviour

Exception is thrown `No current transaction present. Consider declaring @Transactional on the surrounding method`.

Is it intended? Why is there this limitation?

**Comment by graemerocher** [comment]
If you preform multiple queries with `@Transactional` you would get multiple transactions which is not really efficient

**Comment by morki** [comment]
@graemerocher thank you for feedback. I think I dont understand you. Every query inside method annotated with `@Transactional` will get its own transaction?

**Comment by morki** [comment]
@graemerocher it seems `@javax.transaction.Transactional` does not work in native image.  If using `@TransactionalAdvice` or `@Readonly`, everythink works as expected even in native image.

For curiosity, even if I change  `@Transactional` to `@TransactionalAdvice` only on one method, everything works fine, too, even the other methods.

On classic JVM, both cases are ok. Maybe I am doing something wierd here.

**Comment by graemerocher** [comment]
Seems like a bug. I will take a look

**Comment by graemerocher** [comment]
I can't seem to reproduce this, could you provide an example that reproduces the issue?

**Comment by morki** [comment]
Of course, working on it.

**Comment by morki** [comment]
@graemerocher so, as a base, I took GraalVM test for JOOQ that I made for @ilopmar to test JOOQ in  `micronaut-graal-tests`.

Without transactions: 
- https://github.com/morki/micronaut-data-jdbc-graal/tree/jooq-diff
- working fine

After adding Micronaut Data TX with `@Transactional`:
- https://github.com/morki/micronaut-data-jdbc-graal/tree/tx-not-working
- working on JVM but not in native image
- exception: `No current transaction present. Consider declaring @Transactional on the surrounding method`

After replacing `@Transactional` with `@TransactionalAdvice`:
- https://github.com/morki/micronaut-data-jdbc-graal/tree/tx-working
- working on JVM and in native image again

**Comment by graemerocher** [comment]
Thanks will have a look

**Comment by graemerocher** [comment]
The issue is fixed in core and will be corrected in the next 2.0 milestone release

**Comment by morki** [comment]
Thank you very much :)
